### ISSUE TITLE
Xcode 7 Compiler Problems

### ISSUE THREAD

**Issue by pr1001** [CocoaPods/CocoaPods#4202]
I just upgraded to Xcode 7 and CocoaPods 0.38.2 and, after resolving some Swift 2 issues, got the following linker error with my mixed Objective-C & Swift project targeting iOS 7:

	Undefined symbols for architecture x86_64:
	  "_OBJC_CLASS_$_RKManagedObjectRequestOperation", referenced from:
	      objc-class-ref in NetworkHelper.o
	  "_OBJC_CLASS_$_RKManagedObjectResponseMapperOperation", referenced from:
	      type metadata accessor for ObjectiveC.RKManagedObjectResponseMapperOperation in MappingHelper.o
	  "_RKLogCoreDataError", referenced from:
	      ___33-[RKEntityByAttributeCache load:]_block_invoke in libRestKit.a(RKEntityByAttributeCache.o)
	      ___43-[RKManagedObjectImporter finishImporting:]_block_invoke in libRestKit.a(RKManagedObjectImporter.o)
	ld: symbol(s) not found for architecture x86_64
	clang: error: linker command failed with exit code 1 (use -v to see invocation)

CoreData is included in my .pch file and is included before RestKit in the bridging header, and Bitcode is off. I'm using RestKit 0.24.1.

Since it appears that RestKit isn't detecting the presence of the CoreData framework and its conditional declarations are accordingly getting passed over, I've tried lots of various ways to tell my app or the RestKit pod about the presence of Core Data, but nothing seems to help.

I created an [issue with RestKit](https://github.com/RestKit/RestKit/issues/2303), but I think this may be a CocoaPods + Xcode issue, as I tried using CocoaPods 0.39.0.beta.4 and now get a different error:

    /Users/perobinett/Lua/lua.ios.iphone/Pods/Headers/Public/RestKit/ObjectMapping.h:21:9: error: 'RKObjectMapping.h' file not found
    #import "RKObjectMapping.h"
            ^
    <unknown>:0: error: failed to import bridging header '/Users/perobinett/Lua/lua.ios.iphone/Lua/Lua-Bridging-Header.h'

I also created a [question on Stack Overflow](http://stackoverflow.com/q/32619935/46768).

**Comment by segiddins** [comment]
There's an open issue about using restkit as a static lib with the latest CocoaPods version.

**Comment by pr1001** [comment]
Which issue? Do previous CocoaPods versions support it? I tried downgrading to CocoaPods 0.38.1, where compilation hangs on RestKit issues, while 0.38.0 has the same issues as 0.38.2.

**Comment by pr1001** [comment]
I got around my issue in the short term by sticking with the CocoaPods 0.37.2 I had installed earlier and using `xcode-select` to run against Xcode 6.

**Comment by segiddins** [comment]
It's an issue with RestKit, not CocoaPods.

**Comment by ewhite-dev** [comment]
@segiddins can you look into why the beta of cocoapods breaks restkit? What is the RestKit package doing wrong/incompatible with the beta, since it worked for all previous versions of cocoapods? What is the fix to make it compatible

**Comment by segiddins** [comment]
I'm the developer of RestKit, and the issue with it is that it doesn't use nested imports, as it should.

**Comment by ewhite-dev** [comment]
@segiddins is there something I can do locally? Like remove it from my pods file and clone it from your repo? Or is there a patch in pods etc

I'm not currently too familiar with the nuances of ruby, pods, and its packages, as they have just been working till now

**Comment by segiddins** [comment]
No, restkit needs to be patched, which is waiting on either a PR or me to get a computer

**Comment by alexszilagyi** [comment]
@segiddins : can you confirm to me is is related to [this question](http://stackoverflow.com/questions/32712939/restkit-rkserialization-h-file-not-found-via-cocapods) also?

**Comment by ewhite-dev** [comment]
@segiddins any possibility of an ETA? My project uses this for network calls and serialization, and had to be upgraded for ios 9 which revealed this incompatibility

**Comment by segiddins** [comment]
No eta -- I don't use RestKit personally any more, and I don't have a computer to develop on at the moment. Your best bet of getting s fix will be to make a PR to RestKit yourself.

**Comment by ewhite-dev** [comment]
@segiddins yeah I can make a pull request, my main question has been what exactly do I need to fix? Something about nested imports is broken, but what is that?

**Comment by AcidicSkittles** [comment]
@ewhite-dev I'm facing the same problem. After finding http://stackoverflow.com/a/32582264/3831365 I suppose the problem is it's not finding the headers recursively anymore. A temporary work around is to modify the "Header Search Paths" in your target settings and set everything to "recursive".

**Comment by ewhite-dev** [comment]
@AcidicSkittles that worked!

For anyone else, all of the headers have to be set to recursive in all of your targets, not just RestKit.

Also you have to use Cocoapods 0.39.0.beta.4 to successfully Archive

**Comment by alexszilagyi** [comment]
@ewhite-dev : can you post a screen-shot of that? or paste the value that you've entered. I assume that is more like: `../../Headers/`

**Comment by ewhite-dev** [comment]
@alexszilagyi in each target, go to build settings, make sure you are seeing ALL instead of Basic settings. Go to the Header Search Paths settings. Double click on the value, and a popup shows you all your header paths, and next to each one it says recursive or non-recursive, you can change which one it is by click the drop down.

**Comment by tomerhy** [comment]
Problem is, that RestKit version (0.22.0) specified in the Podfile is not compatible with Cocoapods 0.38.x. Switching to latest version 0.24.1 should fix this:

In Podfile:

pod 'RestKit', '0.24.1'

**Comment by tommulder** [comment]
I am now using Restkit V0.25 and cocoapods V0.39.0.beta.4

I have an Xcode workspace with an Embedded Framework project and my main app project + pods project. 

I was using a previous version of pods but that was causing Generic Xcode archive issues (XCode 7) -> the embedded framework was being included in the xcode archive. After updating to the latest cocoapods (beta.4) I had issues with the embedded framework project not finding Restkit headers. I set all the header search paths (for pods modules) to recursive in the Embedded Framework project and that seemed to fix that problem. After that, headers were being included in the Xcode archive so was still getting generic archives. For each pod target in the pods project in the "Headers" build phase, I shifted all the headers from public to Project, then I could build an IOS archive without any compiling, linking errors.
